src/PublicKey.ts

```diff
@@ -26,14 +33,35 @@ export class PublicKey {
         const x = publicKey.getPublic().getX().toArray('be', 32);
         const y = publicKey.getPublic().getY().toArray('be', 32);
         if (!ec) {
-            ec = constructElliptic(keyType);
+            if (keyType === KeyType.k1) {
+                ec = new EC('secp256k1');
+            } else {
+                ec = new EC('p256');
+            }
         }
         return new PublicKey({
             type: keyType,
             data: new Uint8Array([(y[31] & 1) ? 3 : 2].concat(x)),
         }, ec);
     }
 
+    /** Instantiate public key from a `CryptoKey`-format public key */
+    public static async fromWebCrypto(publicKey: CryptoKey, ec?: EC): Promise<PublicKey> {
+        if (publicKey.extractable === false) {
+            throw new Error('Crypto Key is not extractable');
+        }
+        if (!ec) {
+            ec = new EC('p256');
+        }
+
+        const extractedArrayBuffer = await crypto.subtle.exportKey('spki', publicKey);
+        const extractedDecoded = arrayToString(extractedArrayBuffer);
+        const derHex = Buffer.from(extractedDecoded, 'binary').toString('hex');
+        const publicKeyHex = derHex.replace('3059301306072a8648ce3d020106082a8648ce3d030107034200', '');
+        const publicKeyEc = ec.keyFromPublic(publicKeyHex, 'hex');
+        return PublicKey.fromElliptic(publicKeyEc, KeyType.r1, ec);
```

Comment:
Only p256 is supported for the Web Crypto API but it is possible that someone could put a k1 `ec` object and the code would break.  I think a check is likely needed, I'll add it.

Reply:
Actually, turns out the `elliptic` library does not easily label the curve utilized in the object.  It would not be as simple as checking `ec.curve.label = 'p256';`.  I will remove the ability of providing an `ec` object to the functions.

Reply:
Thanks! This looks more clear too me.